core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -59,7 +59,11 @@ public enum Codec {
 
     public static Codec fromName(String codecName) {
       Preconditions.checkArgument(codecName != null, "Codec name is null");
-      return Codec.valueOf(codecName.toUpperCase(Locale.ENGLISH));
+      try {
+        return Codec.valueOf(codecName.toUpperCase(Locale.ENGLISH));
```

Comment:
I noticed that we use a mixture of `Locale.ENGLISH` and `Locale.ROOT` across the codebase. Should this be potentially unified @rdblue?

Reply:
I'm not sure when we should use Locale.ROOT, but I do know it is generally good practice.

We use ENGLISH in some places where we know that the word will be an English word, like this case.